pkgs/build-support/setup-hooks/separate-debug-info.sh

```diff
@@ -16,28 +16,40 @@ _separateDebugInfo() {
     # Find executables and dynamic libraries.
     local i
     while IFS= read -r -d $'\0' i; do
-        if ! isELF "$i"; then continue; fi
-
-        # Extract the Build ID. FIXME: there's probably a cleaner way.
-        local id="$($READELF -n "$i" | sed 's/.*Build ID: \([0-9a-f]*\).*/\1/; t; d')"
-        if [ "${#id}" != 40 ]; then
-            echo "could not find build ID of $i, skipping" >&2
-            continue
+        if isELF "$i"; then
+
+            # Extract the Build ID. FIXME: there's probably a cleaner way.
+            local id="$($READELF -n "$i" | sed 's/.*Build ID: \([0-9a-f]*\).*/\1/; t; d')"
+            if [ "${#id}" != 40 ]; then
+                echo "could not find build ID of $i, skipping" >&2
+                continue
```

Comment:
maybe we should strip .so without Build IDs as well, they could be unexpected bloat.

Reply:
:thinking: Yes, interesting. The more I think about this, the more it feels what we should actually be doing is falling through to allow the regular strip setup hook to do its job afterwards instead of disabling it with `dontStrip`. Because surely now all the outputs other than `debug` _are_ safe to strip. Should this setup hook perhaps just concern itself with pulling the debug info out to `debug` and leave the existing stripping mechanisms to continue operating as normal?

Though that would be a bigger change and I'm not confident I'd understand all the ramifications of that.